client/packages/common/src/ui/forms/JsonForms/components/Array.tsx

```diff
@@ -0,0 +1,189 @@
+            onClick={addItem(
+              path,
+              uischema.defaultNewItem ?? createNewItem(foundUISchema)
+            )}
+          />
+        </Box>
+      </Box>
+      {data.map((child, index) => {
+        return (
+          <Accordion
+            key={index}
+            defaultExpanded={index === data.length - 1}
+            sx={{
+              mt: '0 !important',
+              mb: index === data.length - 1 ? '20px !important' : 1,
+            }}
+          >
+            <AccordionSummary
+              expandIcon={<ChevronDownIcon />}
+              sx={{
+                '&:hover .array-remove-icon': { visibility: 'visible' },
+                margin: '0 !important',
+                minHeight: '0 !important',
+                '.MuiAccordionSummary-content': {
+                  margin: '5px !important',
+                },
+                '.Mui-expanded': {
+                  marginBottom: '0 !important',
+                },
+              }}
+            >
+              <Box
+                display="flex"
+                width={FORM_LABEL_COLUMN_WIDTH}
+                justifyContent="space-between"
+                alignItems="center"
+              >
+                <IconButton
+                  icon={<MinusCircleIcon />}
+                  label={t('label.remove')}
+                  color="primary"
+                  className="array-remove-icon"
+                  sx={{ visibility: 'hidden' }}
+                  onClick={removeItems(path, [index])}
+                />
+                <Typography
+                  sx={{
+                    fontWeight: 'bold',
+                    textAlign: 'end',
+                    whiteSpace: 'nowrap',
+                  }}
+                >
+                  {uischema?.itemLabel
+                    ? RegexUtils.formatTemplateString(uischema?.itemLabel, {
+                        ...child,
+                        index: index + 1,
+                      })
+                    : index + 1}
+                </Typography>
+              </Box>
+            </AccordionSummary>
+            <AccordionDetails>
+              <JsonFormsDispatch
+                key={index}
+                schema={schema}
+                uischema={foundUISchema}
+                enabled={enabled}
+                path={`${path}.${index}`}
+              />
+            </AccordionDetails>
+          </Accordion>
+        );
+      })}
+    </Box>
+  );
+};
+
+const withJsonFormsArrayCustomProps = (
+  component: ComponentType<ArrayControlCustomProps>
+) =>
+  withJsonFormsArrayControlProps(component as ComponentType<ArrayControlProps>);
+
+export const Array = withJsonFormsArrayCustomProps(UIComponent);
+
+const createNewItem = (foundUISchema: FoundUiSchema) => {
+  try {
+    const fields = foundUISchema.elements.map(e => [
+      e.scope.split('/').pop(),
+      undefined,
+    ]);
+    return Object.fromEntries(fields);
+  } catch {
+    return {};
+  }
+};
```

Comment:
It wasn't happy when a new `null` element was added to data array, so this just constructs an object with the correct keys (but no values), which seems to work ok.

Reply:
Hmm, now that I've fixed the re-rendering problem with a4511fd06, this doesn't seem necessary now -- I can just make the new array element `null` and it works fine (it creates the object and fields one by one as they are filled in). 

Wondering if I should remove it, or leave it just in case it becomes useful in #174 ?

Might leave it here for now, but I'll bear it in mind when we get to #174 and I'll remove it if possible.